src/server/webhookServer.ts

```diff
@@ -0,0 +1,74 @@
+import * as http from 'http';
+import _debug from 'debug';
+import { LIBRARY_NAME } from '../constants';
+import { IAlice } from '../alice';
+
+const debug = _debug(LIBRARY_NAME);
+
+export interface IWebhookServerConfig {
+  port: number;
+  webhookUrl: string;
+  options: object;
+  handleRequest: IAlice['handleRequest'];
+}
+export interface IWebhookServer {
+  start(): void;
+  stop(): void;
+}
+
+export class WebhookServer {
+  private server: http.Server;
+  private port: number;
+  private webhookUrl: string;
+  private isServerStarted: boolean;
+  private handleRequest: IAlice['handleRequest'];
+
+  constructor(config: IWebhookServerConfig) {
+    this.port = config.port;
+    this.webhookUrl = config.webhookUrl;
+    this.handleRequest = config.handleRequest;
+    this.isServerStarted = false;
+
+    debug(`create server on: ${this.port}, ${this.webhookUrl}`);
+    this.server = http.createServer(
```

Comment:
I think we can split this function one more time (something like this):
```
constructor {
    this.server = http.createServer(
        (request, response) => this._handleRequest(request, response));
}

private async _handleRequest(request, response) {
   const requestData = await AliceHttpServer._readRequest(request);
   const responseData = await this.handleRequestData_(requestData);
   await AliceHttpServer._sendResponse(response, responseData);
}

private static async _sendResponse(response, responseBody: IApiReposne): Promise {
   response.statusCode = 200;
   response.setHeader('Content-Type', 'application/json');
   response.end(JSON.stringify(responseBody));
}

private static async _readRequest(request): Promise<IApiRequest> {
    return new Promise((resolve, reject) => {
      const body: Array<Buffer> = [];
      request
            .on('data', chunk => body.push(chunk))
            .on('end', () => {
                if (request.method !== 'POST' || request.url !== this.webhookUrl) {
                   response.statusCode = 400;
                   response.end();
                   reject();
                }
                resolve(JSON.parse<IApiRequest>(requestData));
            })
            .on('error', reject); // I don't know if there is something like this.
    });
}
```

Reply:
Splitted into separate functions and refactored